### PR TITLE
Add port to host header for signing.

### PR DESCRIPTION
**Pull request by krisis**
Also fixed issue in fetching endpoint corresponding to region for AWS S3.

### CODE REVIEW

File: src/Network/Minio/API.hs [hunk 1]
```diff
@@ -104,20 +106,38 @@ buildRequest ri = do
...
+        return $ connectHost ci
+      Just r -> do
+        if "amazonaws.com" `T.isSuffixOf` (connectHost ci)
+          then do
+          maybe
+               (throwM $ ME $ ValidationError $ MErrVRegionNotSupported r)
+               return
+               (Map.lookup r awsRegionMap)
+          else do
+          return $ connectHost ci
```
Comment (donatello) [inline src/Network/Minio/API.hs]: All the 4 `do` terms are extraneous as they contain just one statement inside. The `case region of` should also begin on the same line as `regionHost`.

File: src/Network/Minio/Sign/V4.hs [hunk 2]
```diff
@@ -98,7 +98,9 @@ signV4AtTime ts ci ri =
     outHeaders = authHeader : headersWithDate
     timeBS = awsTimeFormatBS ts
     dateHeader = (mk "X-Amz-Date", timeBS)
-    hostHeader = (mk "host", encodeUtf8 $ connectHost ci)
+    hostHeader = (mk "host", encodeUtf8 $
+                   format "{}:{}" $
+                   [connectHost ci, show $ connectPort ci])
```
Comment (donatello) [inline src/Network/Minio/Sign/V4.hs]: Break lines only when the are too wide. Here it looks like the the line with `format` can be joined with the previous line.

Comment (krisis) [inline src/Network/Minio/Sign/V4.hs]: Done

File: src/Network/Minio/API.hs (hunk 1 above)
Comment (krisis) [inline src/Network/Minio/API.hs]: Done.

**Pull request merged**